### ISSUE TITLE
The prediction became poor after upgrading from 1.0 to 2.* for R

### ISSUE THREAD

**Issue by zhangsssss** [Microsoft/LightGBM#1202]
## Environment info
C++/Python/R version: R

## examples 
The R version 1.0 works:
bst = lgb.train(params, dTrain, 150, list(train = dTrain,test = dTest))
the last row of the output is 
[150]: train's rmse:0.0134871  test's rmse:0.015928 
when calculating the test set error by hand:
pred = predict(bst, data_sparse[test_ind,]) # it takes sometime for large data, but the distribution looks good
sqrt(sum(dataRaw$weight[test_ind]*(dataRaw$y[test_ind] - pred)^2)/sum(dataRaw$y[test_ind]))
[1] 0.015928   # which matches the last row of model training output
summary(pred) 
     Min.        1st Qu.    Median      Mean   3rd Qu.      Max. 
-0.001950  0.009145  0.015400  0.019230  0.024810  0.301900  

The R version 2.* doesn't work
bst = lgb.train(params, dTrain, 150, list(train = dTrain,test = dTest))
the last row of the output is 
[150]:    train's rmse:0.0133429  test's rmse:0.0156751
when calculating the test set error by hand:
pred = predict(bst, data_sparse[test_ind,])  # it is super fast even for large data, however, seems it tend to predict almost everything to be the same （Q1 = Q2 = Q3）
sqrt(sum(dataRaw$weight[test_ind]*(dataRaw$y[test_ind] - pred)^2)/sum(dataRaw$y[test_ind]))
[1] 0.0244318   # which DOSEN'T matches the last row of model training output
summary(pred) 
     Min.        1st Qu.    Median      Mean   3rd Qu.      Max. 
 -0.0006544  0.0060730  0.0060730  0.0076750  0.0060730  0.2611000

**Comment by guolinke** [comment]
@zhangsssss 
can you provide more information ?
1. more information about the train/test data, include its dimension, type(matrix, csc, ..) 
2. a reproduce example by using random data 
3. the full script

**Comment by Laurae2** [comment]
@guolinke probably due to the default hyperparameter changes.

**Comment by guolinke** [comment]
@Laurae2 I think the parameter changing doesn't effect the prediction. 
In this case, the metric output in training and prediction are different. 
@zhangsssss can you provide a reproduce example?

**Comment by zhangsssss** [comment]
Hi I'm so sorry for the late reply. As I installed the latest version (lightgbm_2.1.0.tar.gz), the problem seems to have been gone... The versions I installed on around 7/15/2017 and 12/01/2017 had the problem I mentioned. Both times I just followed your installation guidance so the version should be the latest at those moments. Now my VM with these old versions of LightGBM is gone, so I'm not able to provide more info and I can't reproduce the case. Sorry about that... but thankfully it's good now!
